release-tools/Makefile.prog

```diff
@@ -26,6 +26,6 @@ install:${PROG}
 clean:
 	rm -f ${PROG} ${SRCS:.c=.o}
 
-${PROG}:${SRCS:.c=.o}
-	${CC} -o ${PROG} ${SRCS:.c=.o} ${LDFLAGS} ${LDADD_EXTRA} ${LDADD_REQ} ${LDADD_POSIX}
+${PROG}:${SRCS:.c=.o} ${PROJECT_LIBS}
```

Comment:
Don't you need to add code to `Makefile.prog` telling it how to build `${PROJECT_LIBS}`?

Reply:
That was my first thought as well, but then I realized that I could rely on the "loop-back" building, just like `cpusupport-config.h`.

Namely, when you type `make` in a subdirectory:
- it jumps back to the top level and does `make buildsubdir` (with some variables set)
- `buildsubdir` checks if it needs build `cpusupport-config.h` etc.
- `buildsubdir` also unconditionally does `cd libspiped && make all`, which ensures that `libspiped.a` is up to date.
- then it returns to the subdirectory and does `make $PROG`.

(as a side note, I just realized that $PROJECT_LIBS won't work with `make -f Makefile.BSD` as it is currently, but it only needs a small fix.)

((even after that fix, `make -f Makefile.BSD` won't track libspiped.a as a dependency, but hopefully that's ok.))

Reply:
Ah, got it.  So effectively `HAVE_BUILD_FLAGS` is `HAVE_BUILD_FLAGS_AND_LIBFILE_IS_UP_TO_DATE`.  (No, I'm not suggesting that the variable name gets changed.)